src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/CustomAttributeBasedDependencyAlgorithm.cs

```diff
@@ -137,25 +137,6 @@ private static void AddDependenciesDueToCustomAttributes(ref DependencyList depe
                         dependencies.AddRange(caDependencies);
                         dependencies.Add(factory.CustomAttributeMetadata(new ReflectableCustomAttribute(module, caHandle)), "Attribute metadata");
                     }
-
-                    // Works around https://github.com/dotnet/runtime/issues/81459
-                    if (constructor.OwningType is MetadataType { Name: "EventSourceAttribute" } eventSourceAttributeType)
-                    {
-                        foreach (var namedArg in decodedValue.NamedArguments)
-                        {
-                            if (namedArg.Name == "LocalizationResources" && namedArg.Value is string resName
-                                && InlineableStringsResourceNode.IsInlineableStringsResource(module, resName + ".resources"))
-                            {
-                                dependencies ??= new DependencyList();
-                                var accessorMethod = module.GetType(
-                                    InlineableStringsResourceNode.ResourceAccessorTypeNamespace,
-                                    InlineableStringsResourceNode.ResourceAccessorTypeName)
-                                    .GetMethod(InlineableStringsResourceNode.ResourceAccessorGetStringMethodName, null);
-                                dependencies.Add(factory.ReflectedMethod(accessorMethod), "EventSource used resource");
-                            }
-                        }
-                    }
-                    // End of workaround for https://github.com/dotnet/runtime/issues/81459
```

Comment:
We don't still want this for cases where a 3rd-party uses LocalizationResources?

Reply:
We don't remove resources in user assemblies so this should not be needed for those unless the user does the same gestures to get them removed.